packages/core-commands/src/index.js

```diff
@@ -1 +1,44 @@
+/**
+ * WordPress dependencies
+ */
+import { createRoot, StrictMode } from '@wordpress/element';
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+import { CommandMenu } from '@wordpress/commands';
+
+/**
+ * Internal dependencies
+ */
+import { useAdminNavigationCommands } from './admin-navigation-commands';
+import { useSiteEditorNavigationCommands } from './site-editor-navigation-commands';
+import { unlock } from './lock-unlock';
 export { privateApis } from './private-apis';
+
+const { RouterProvider } = unlock( routerPrivateApis );
+
+// Register core commands and render the Command Palette.
+function CommandPalette() {
+	useAdminNavigationCommands();
+	useSiteEditorNavigationCommands();
+	return (
+		<RouterProvider pathArg="p">
```

Comment:
Some core commands use the `useHistory` hook. This `routerProvider` is required for these commands to work correctly.